src/memray/_memray/tracking_api.cpp

```diff
@@ -782,21 +799,23 @@ Tracker::computeMainTidSkip()
     return num_frames - 1;
 }
 
-void
-Tracker::trackAllocationImpl(void* ptr, size_t size, hooks::Allocator func)
+void __attribute__((noinline)) Tracker::fillNativeTraceIfNeeded(NativeTrace& trace)
 {
-    if (RecursionGuard::isActive || !Tracker::isActive()) {
-        return;
+    if (PythonStackTracker::s_native_tracking_enabled) {
```

Comment:
We should bubble up this conditional to make the call go away

Reply:
I have modify this by adding a static function to the tracker that acts as a proxy to `PythonStackTracker::s_native_tracking_enabled` so we can move the conditional up. That will also help with #294 

The main difference is that the compiler can indeed inline that call now